types/address.go

```diff
@@ -569,11 +571,12 @@ func (ca ConsAddress) String() string {
 	}
 
 	var key = conv.UnsafeBytesToStr(ca)
+	consAddrMu.Lock()
```

Comment:
will this slow down the cache? Would a read lock make more sense?

Reply:
@robert-zaremba ?

Reply:
Yes, good point. I used the same, lock as the one from https://github.com/cosmos/cosmos-sdk/pull/8694
I will create another change for that.

Reply:
sync.RWMutex is going to give surprising results for situations unless you have multiple readers and a single producer scenario plus overwhelming evidence that reads are slower, otherwise one is tacking on expenses that are unnecessary. Please see https://gist.github.com/odeke-em/234bc88fbaf9e439b70a618ce6674659#file-aresults-md

I’d just stick to entirely using sync.Mutex unless the mutex is being held for a long time. In fact there are plans to fix sizes and cost of sync.RWMutex like golang/go#37142